src/plugins/discover/public/application/main/state_management/discover_state.ts

```diff
@@ -412,17 +417,24 @@ export function getDiscoverStateContainer({
     }
   };
 
+  const setIsLoading = (value: boolean) => {
+    internalStateContainer.transitions.setIsLoading(value);
+  };
+
   const onDataViewEdited = async (editedDataView: DataView) => {
+    setIsLoading(true);
```

Comment:
For me it appears we just use `isDataViewLoading` in 

https://github.com/elastic/kibana/blob/8545b9ccfbad97881406e56ffd96f452c94032b8/src/plugins/discover/public/application/main/components/layout/discover_documents.tsx#L169

and 

https://github.com/elastic/kibana/blob/73694426f943f00f74556406dcb26c66eb4a772a/src/plugins/discover/public/application/main/components/layout/discover_layout.tsx#L104-L107

We set it to true when a data view is being changes. 

With the new `isLoading` prop those components should not be displayed. So I'd say it's not necessary ... one think I wonder, would it make sense to combing all those loading states into one prop? one that can be used to get information about general loading state, data view loading state, data loading state ... just load thinking

Reply:
Yeah, I was also wondering if we could use `isDataViewLoading` instead of introducing a new `isLoading`. 
Editing a data view might be considered as "loading" its new state.

Reply:
So I splitted this work (again): 
Hopefully near term solution: #199982
Experimental loading refactoring, moving to one loading state instead of `isDataViewLoading` and `isLoading`: #199956